README.md

```diff
@@ -172,21 +172,54 @@ app.listen(80, function () {
+
+// `mismatchContinue: false` will work with any of the `origin` options supported
+// by this module. However it won't help prevent CSRF unless you whitelist specific
+// origins. Also see caveats below!
+var whitelist = ['https://same-origin.com', 'https://other-origin.com'];
+
+app.del('/products/:id', cors({ origin: whitelist, mismatchContinue: false }), function (req, res, next) {
+  res.json({msg: 'This will not be executed if called from https://evil-origin.com.'});
+});
+
+app.listen(80, function () {
+  console.log('CORS-enabled and CSRF-protected web server listening on port 80');
+});
+```
+
+Make sure to whitelist cross-origin AND same-origin requests, since the middleware will reject any request whose origin is not on the list. This also means that the middleware will reject requests with _missing_ origins. This means that `mismatchContinue: false` is NOT SUITED for routes where the `Origin` header may be missing, e.g. routes loaded directly by users in the browser, or routes loaded by `<img>` tags. However AJAX requests should [always have `Origin` headers](https://bugs.chromium.org/p/chromium/issues/detail?id=512836#c3).
+
+If you don't want to enable a route for cross-origin usage but you _do_ wish to verify same-origin usage, you can apply this module to that route with _only_ the same origin whitelisted.
+
 ## Configuration Options
 
 * `origin`: Configures the **Access-Control-Allow-Origin** CORS header. Possible values:
   - `Boolean` - set `origin` to `true` to reflect the [request origin](http://tools.ietf.org/html/draft-abarth-origin-09), as defined by `req.header('Origin')`, or set it to `false` to disable CORS.
   - `String` - set `origin` to a specific origin. For example if you set it to `"http://example.com"` only requests from "http://example.com" will be allowed.
   - `RegExp` - set `origin` to a regular expression pattern which will be used to test the request origin. If it's a match, the request origin will be reflected. For example the pattern `/example\.com$/` will reflect any request that is coming from an origin ending with "example.com".
   - `Array` - set `origin` to an array of valid origins. Each origin can be a `String` or a `RegExp`. For example `["http://example1.com", /\.example2\.com$/]` will accept any request from "http://example1.com" or from a subdomain of "example2.com".
-  - `Function` - set `origin` to a function implementing some custom logic. The function takes the request origin as the first parameter and a callback (which expects the signature `err [object], allow [bool]`) as the second.
+  - `Function` - set `origin` to a function implementing some custom logic. The function takes the request origin as the first parameter and a callback (which expects the signature `err [object], allow [bool]`) as the second. If the function calls back with `allow === false`, CORS will not be enabled for the response, but the next handler will still run unless `mismatchContinue === false`.
```

Comment:
As described [below](https://github.com/expressjs/cors/pull/110/files#r108100148), this is merely documentation of pre-existing behavior.